state/machine.go

```diff
@@ -1832,3 +1832,76 @@ func (m *Machine) PendingActions() ([]Action, error) {
 func (m *Machine) RunningActions() ([]Action, error) {
 	return m.st.matchingActionsRunning(m)
 }
+
+// UpdateMachineSeries updates the series for the Machine.
+func (m *Machine) UpdateMachineSeries(series string, force bool) error {
+	buildTxn := func(attempt int) ([]txn.Op, error) {
+		if attempt > 0 {
+			if err := m.Refresh(); err != nil {
+				return nil, errors.Trace(err)
+			}
+		}
+		// Exit early if the Machine series doesn't need to change.
+		if m.Series() == series {
+			return nil, jujutxn.ErrNoOperations
+		}
+
+		principals := m.Principals() // unit names
+		verifiedUnits, err := m.verifyUnitsSeries(principals, series, force)
+		if err != nil {
+			return nil, err
+		}
+
+		ops := []txn.Op{{
+			C:      machinesC,
+			Id:     m.doc.DocID,
+			Assert: isAliveDoc,
+			Update: bson.D{{"$set", bson.D{{"series", series}}}},
+		}}
+		ops = append(ops, txn.Op{
+			C:      machinesC,
+			Id:     m.doc.DocID,
+			Assert: bson.D{{"principals", principals}},
```

Comment:
Is this sufficient? What about if a new subordinate is added? The assertions need to match the business logic.